R/toolISOhistorical.R

```diff
@@ -152,25 +152,37 @@ toolISOhistorical <- function(m, mapping = NULL, additional_mapping = NULL, over
                              mapping[is.element(mapping$toISO, a$toISO), c("fromISO", "toISO")],
                              weight = weight + 1e-12,
                              negative_weight = "allow")
+        
+        # fill data
+        if (overwrite) {
+          m[a$toISO, subTime, ] <- mTr
+        } else {
+          m[a$toISO, subTime, ][is.na(m[a$toISO, subTime, ])] <- mTr[is.na(m[a$toISO, subTime, ])]
+        }
+        
+        # delete old lines
+        for (b in a$fromISO) {
+          if (is.element(b, getItems(m, dim = 1.1))) {
+            m <- m[-which(getItems(m, dim = 1.1) == b), , ]
+          }
+        }
         ## aggregation of countries
       } else {
         mTr <- toolAggregate(m[a$fromISO, subTime, ],
                              mapping[is.element(mapping$toISO, a$toISO), c("fromISO", "toISO")], weight = NULL)
-      }
-      # fill data
-      if (overwrite) {
+        
+        # fill data, here no option to not overwrite old data as countries are aggregated
         m[a$toISO, subTime, ] <- mTr
-      } else {
-        m[a$toISO, subTime, ][is.na(m[a$toISO, subTime, ])] <- mTr[is.na(m[a$toISO, subTime, ])]
+        
+        # delete old lines
+        for (b in a$fromISO) {
+          if ((is.element(b, getItems(m, dim = 1.1))) & (!(is.element(b, a$toISO)))) { # only delete lines that are not resulting countries
```

Comment:
Did you get a linter warning here? In if statements `&&` should be used over `&`, as `&&` always results in a single TRUE/FALSE, while `&` can create a vector of TRUE/FALSE which does not work for `if`.
```suggestion
          if ((is.element(b, getItems(m, dim = 1.1))) && (!(is.element(b, a$toISO)))) { # only delete lines that are not resulting countries
```

Reply:
I actually did not for some reason, though that sounds good to me. Maybe it makes sense to use the 'in' syntax as above here as well?